CONTRIBUTING.md

```diff
@@ -217,27 +217,122 @@ $ export THEIA_PLUGIN_ENDPOINT_DISCOVERY_PORT="2506"
 $ yarn theia start /projects/theia-projects-dir --hostname=0.0.0.0 --port=3010
 ```
 
-## How to develop Che Theia remote plugin mechanism
-
-_Please note, this section provides a flow how to develop remote plugin mechanism in Che Theia, but not a remote plugin._
-
-First, create a workspace from prepared [devfile](https://github.com/eclipse/che-theia/blob/master/extensions/eclipse-che-theia-plugin-remote/devfile.yaml), which could be found in the `eclipse-che-theia-plugin-remote` extension folder.
-When workspace is ready:
- - Init Che Theia. This could be done with `che-theia init` command in `/projects/theia` folder or run the init command.
-   Che Theia sources will be awailable at `/projects/theia/che/che-theia`.
- - Now one may make changes in Che Theia remote plugin mechanism in both (Che Theia and Remote plugin) sides.
- - Build Che Theia from `theia-dev` container by executing `yarn` in `/projects/theia` folder or by running corresponding command.
- - Put binaries (*.theia or *.vsix) of your remote plugin(s) into `/projects/remote-plugins/` directory.
-   Note that the plugin(s) shouldn't have any external dependencies.
-   For example, [this sample plugin](https://github.com/eclipse/che-theia-samples/tree/master/samples/hello-world-backend-plugin) might be used.
-   Or a user may generate one using `Generate Hello World plugin package` command.
- - Run dev Che Theia and Remote plugins endpoint in `theia-dev` and `theia-remote-runtime-dev` containers correspondingly.
-   One may use predefined commands to start them.
- - Open `theia-dev` route from `My Workspace` panel and test chenges.
-
-Also it is possible to run watchers for remote plugin mechanism.
-In `theia-dev` container run `npx run watch @eclipse-che/theia-remote` from `/projects/theia` folder to recompile the extension on changes made.
-Also run `yarn watch` in `/projects/theia/examples/assembly` to bring the changes to Che Theia binaries.
-If needed one may start watchers in plugin API extension: `npx run watch @theia/plugin-ext` from `/projects/theia` directory.
-The commands for these actions are also available.
-But please note, you have to restart server to which changes is made.
+# Working with Plugins in Sidecars
+
+In a real Che workspace, many plugins are run in their own "sidecar" container. The idea is that the container furnishes any "native" dependencies the the plugins needs. For example the container that runs the theia IDE back end does not contain a Java SDK, so in order to run a plugin that needs a Java installation, we'll run the plugin inside a container that has Java installed.
+
+As when working on developing Che-Theia itself, the approach is to run a copy of the Theia back end on a different port than that we use for the self hosting workspace. 
+
+## Setting up the devfile
+
+So, starting with a workspace to develop che-theia, we'll have to add a container that will serve as the sidecar running our plugin.
```

Comment:
This is very nice to show how we could build the devfile. Is the final ready-to-use devfile (for java plugin che-in-che) available somewhere? I think it would make sense to keep this somewhere so contributors would have something ready to use.

Reply:
I have chosen not to provide one: I want people to understand what is happening and to be capable of creating their own devfiles. They will have to anyway if their sidecars have any native dependencies.